### ISSUE TITLE
mistake in player preview

### ISSUE THREAD

**Issue by fabienwang** [stujones11/minetest-3d_armor#37]
the armor.get_preview function checks if skin_mod == "skins"
It shouldn't as the checks are already in the used function armor.get_player_skin

**Comment by LeMagnesium** [comment]
As far as my experience with the code goes :
`armor.get_preview` and `armor.get_player_skin` are used in different situations. `armor.get_player_skin` is used to get the skin chosen by a player using a skin mod, or `character.png`. This texture is applied onto the 3d model of the player. `armor.get_preview` is used to get a 2D representation of a player wearing this skin, or if it couldn't retrieve such a picture, `character_preview.png` (in 3d_armor/textures). It is no mistake how the code was written in `armor.get_preview` as every time it is used, when it returns nil, `character_preview.png` is used, and the `if skin_mod == "skins"` just means that "skins" is the only mod supported to give their preview textures. If it's not the skin_mod, then 3d_armor will use the default texture.

To sum up: There's no problem, these are just two different uses of two different functions.

**Comment by stujones11** [comment]
@LeMagnesium, you sum it up very well although I must admit that the skin mod support has become a bit of a mess. I have gone some way to improve the situation in the 'multiskin' branch (which will eventually become version 0.5.0) but I have to be careful not to break support for mods that rely on the existing api
